### ISSUE TITLE
TFAR Zeus issue can't hear other players when controling AI units.

### ISSUE THREAD

**Issue by irish2016** [michail-nikolaev/task-force-arma-3-radio#1153]
ArmA 3 version 1.64 stable
CBA Version: 3.1.0.160928
TFAR Version: 0.9.12

Basically whenever your in zeus and you want to control an AI unit such as a civilian and to Roleplay and interact with other peoples they can't hear you through TFAR like they could in the previous 0.9.8 version and that's an issue. Also whenever your in control of the AI civilian you can't hear them speak to you either its like TFAR completely mutes everything.

**Comment by michail-nikolaev** [comment]
@dedmen some regress probably?

**Comment by Heartbroken-Git** [comment]
I can confirm the issue, my team and I are experiencing it right now. (ArmA 3 1.64 stable, CBA 3.1.0.160928) However some of the team still had the 0.9.8 plugin on top of the 0.9.12 TFAR install (PlayWithSix doesn't update the plugin part apparently). 
We are in the process of ensuring that everyone uses the 0.9.12 plugin and we'll check again.

**Comment by dedmen** [comment]
Can you provide an example mission with only TFAR and CBA?
I deployed a Workaround in 1.0. You can just add this to your init.sqf
```
["unit", {  //current unit changed (Curator took control of unit)  
if (TFAR_currentUnit != (_this select 0)) then {
  TFAR_currentUnit setVariable ["tf_controlled_unit",(_this select 0)];  
} else {
  TFAR_currentUnit setVariable ["tf_controlled_unit",nil];  
};
}] call CBA_fnc_addPlayerEventHandler;
```
I built this hotfix after a guy on BIF reported the same. Without testing if what he said was true.. But after looking at the code it seemed logical. But then a few other guys on BIF told me it works fine.. Then i tested and indeed. Worked fine for me on 0.9.12. Game Master module with its variable set to the one of a playable unit. I'm not a mission builder maybe i did it different than other people do it and accidentally the only working way.


0.9.8 Plugin works fine with 0.9.12 as long as you don't want to test Teamspeak beta versions.

**Comment by irish2016** [comment]
ive put that code in my initServer.sqf it didn't work? unless ive put it in the wrong sqf

**Comment by dedmen** [comment]
@irish2016 well.. I said to put it into init.sqf.. You put it into initServer.sqf and wonder why it doesn't work? Has to run clientSide.. So init.sqf or initPlayerLocal.sqf

**Comment by LCplSmithenC** [comment]
Will this issue ever be fixed? It is a huge problem for my server.

**Comment by dedmen** [comment]
Oops. Sorry already solved.

**Comment by LCplSmithenC** [comment]
Still doesn't work. Using the latest version of TFAR on my server and on TS and still can't hear other players or talk to other players while remote controlling as Zeus....

    On Wednesday, November 30, 2016 2:22 PM, Dedmen Miller <notifications@github.com> wrote:
 

 Oops. Sorry already solved.—

**Comment by dedmen** [comment]
Yep. Its not fixed in 0.9.12.

**Comment by LCplSmithenC** [comment]
Oh. Sorry. Thought you meant it was fixed by saying the issue was closed. Will keep an eye out for the patch in the future! Its critical to my operations! O7 

    On Wednesday, November 30, 2016 8:52 PM, Dedmen Miller <notifications@github.com> wrote:
 

 Yep. Its not fixed in 0.9.12.

**Comment by dedmen** [comment]
Potential workaround. At init.sqf
```
["unit", {
    //current unit changed (Curator took control of unit)

    if (player != (_this select 0)) then {
        player setVariable ["TFAR_controlledUnit",(_this select 0), true];//This tells other players that our position is different
    } else {
        player setVariable ["TFAR_controlledUnit",nil, true];
    };

    TFAR_currentUnit = (_this select 0);
    };

},true] call CBA_fnc_addPlayerEventHandler;

```

**Comment by LCplSmithenC** [comment]
What init.sqf do I add it to? To the unit's initialization box in game? 

    On Thursday, December 1, 2016 10:46 AM, Dedmen Miller <notifications@github.com> wrote:
 

 Potential workaround. At init.sqf["unit", {
    //current unit changed (Curator took control of unit)

    if (player != (_this select 0)) then {
        player setVariable ["TFAR_controlledUnit",(_this select 0), true];//This tells other players that our position is different
    } else {
        player setVariable ["TFAR_controlledUnit",nil, true];
    };

    TFAR_currentUnit = (_this select 0);
    };

},true] call CBA_fnc_addPlayerEventHandler;

**Comment by dedmen** [comment]
Your missions init.sqf. If you don't know how to.. you could add a trigger that fires once at mission start.

**Comment by LCplSmithenC** [comment]
Ok. I already have a trigger that set's the view distance so ill just add another one with the code 

    On Thursday, December 1, 2016 11:30 AM, Dedmen Miller <notifications@github.com> wrote:
 

 Your missions init.sqf. If you don't know how to.. you could add a trigger that fires once at mission start.—

**Comment by tandkjaer** [comment]
I still have this problem and the init.sqf fix isen't working.. Any help is greatly appriciatet :)

**Comment by Aviditas** [comment]
I also have this problem, the SQF fix doesn't work. Please get this fixed

**Comment by dedmen** [comment]
@Aviditas this is already fixed. That's why this issue is closed. Currently I'd rather work on getting 1.0 released than finding out why my 0.9.12 workaround doesn't work
